pandas/core/frame.py

```diff
@@ -1022,13 +1022,15 @@ def itertuples(self, index=True, name="Pandas"):
         # fallback to regular tuples
         return zip(*arrays)
 
-    def __len__(self):
+    def __len__(self) -> int:
         """
         Returns length of info axis, but here we use the index.
         """
         return len(self.index)
 
-    def dot(self, other):
+    def dot(
+        self, other: Union[Series, DataFrame, np.ndarray]
+    ) -> Union[Series, DataFrame, np.ndarray]:
```

Comment:
can we get ndarray back?  id expect it to always be Series or DataFrame

Reply:
removed ndarray